pkg/apis/v1/ec2nodeclass_conversion.go

```diff
@@ -161,6 +157,16 @@ func (in *EC2NodeClass) ConvertFrom(ctx context.Context, from apis.Convertible)
 				},
 			}}
 		}
+		// If there are no AMISelectorTerms specified, we will fail closed when converting the NodeClass. Users must
+		// pin their AMIs **before** upgrading to Karpenter v1.0.0 if they were using the Ubuntu AMIFamily.
+		// TODO: jmdeal@ verify doc link to the upgrade guide once available
```

Comment:
This needs to be updated, right?
```suggestion
		// If there are no AMISelectorTerms specified, we mark the ec2nodeclass as incompatible.
		// Karpenter will ignore incompatible ec2nodeclasses for provisioning and computing drift.
		// Users need to pin their AMIs **before** upgrading to Karpenter v1.0.0 if they were using the Ubuntu AMIFamily.
```